### PR TITLE
Diagram based Database

### PR DESCRIPTION
**Pull request by EddyShimwa**
### At this step we create a database based on diagram! And we have been to:
- Create table tables like patients, invoices, invoice_items, medical_histories,  and treatments!

### The figure bellow shows how the database schema looks

pair_programming_schema

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    REFERENCES public.patients (id) MATCH SIMPLE
+    ON UPDATE NO ACTION
+    ON DELETE NO ACTION
+    NOT VALID;
+ALTER TABLE IF EXISTS public.medical_histories
+    ADD FOREIGN KEY (id)
+    REFERENCES public.treatments (id) MATCH SIMPLE
+    ON UPDATE NO ACTION
+    ON DELETE NO ACTION
+    NOT VALID;
```
Comment (omar-muhamad) [inline schema_based_on_diagram.sql:70]: - [ ] Please create a new table `medical_histories_treatments` to address the many-to-many relationship between `medical_histories` and `treatments` tables and add the foreign keys in it:
```sql
 CREATE TABLE medical_histories_treatments (
    medical_history_id int references medical_histories(id),
    treatment_id int references treatments(id),
  );
```

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+
+
+
+
+
+
+
```
Comment (omar-muhamad) [inline schema_based_on_diagram.sql]: - [ ] Kindly make sure that all FOREIGN KEYS have indexes created

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @EddyShimwa and @Kashif-Saleem-Ghuman,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Kashif-Saleem-Ghuman** [comment]
@Omar-Muhamad  Thank you for reviewing our PR and providing valuable feedback

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi @Kashif-Saleem-Ghuman and @EddyShimwa     :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by EddyShimwa** [comment]
Thanks @hiromataba for reviewing this PR!!

**Comment by Kashif-Saleem-Ghuman** [comment]
@hiromataba  Thank you for approving our PR :)

**Pull request merged**